admin/tool/urlpreview/index.php

```diff
@@ -38,27 +40,24 @@
     throw new moodle_exception('noguest');
 }
 
-$messageform = new \tool_urlpreview\form\message_form();
-
-
 echo $OUTPUT->header();
 if (isloggedin()) {
     echo '<h3>Welcome, ' . fullname($USER) . '</h3>';
 }
-
-$messageform->display();
-
-if ($data = $messageform->get_data()) {
-
-    // Fetch the URL and get its metadata.
-    $url = $data->url;
-    // Assuming the unfurl_store class is available.
+$templatedata = [
+    'action' => 'index.php',
+    'formClass' => 'form-class',
+    'urlLabel' => 'URL',
```

Comment:
You don't need to pass in language string, you can use them directly inside the template like this:

https://moodledev.io/docs/guides/templates#str

I would also hard code the form class, but keep the action as a template variable

Reply:
actually I would ditch the form-class and its custom style and just add various pre-existing bootstrap classes 

You can view a similar example here on your local:

https://master.localhost/admin/tool/componentlibrary/docspage.php/moodle/components/search-input/#nav-rendered42

or here in code:

https://github.com/moodle/moodle/blob/master/lib/templates/search_input.mustache